tensorflow/core/kernels/string_split_op.cc

```diff
@@ -122,6 +122,84 @@ class StringSplitOp : public OpKernel {
   bool skip_empty_;
 };
 
+class StringUTF8SplitOp : public OpKernel {
```

Comment:
IIUC, the implementation of these two kernels is almost identical, save for the different function call to split (and the vall to validating UTF8 of the delimiter).

Could we refactor the kernel implementation so that a bulk of the code is shared? Something along the lines of:

```c++
class StringSplitOpBase : public OpKernel {
    ...
    Status ValidateDelimiter(const string& delimiter) = 0;
    Status Split(const string& input, const string& delimiter, vector<string>* parts) = 0;
}

class StrignSplitOp : public StringSplitOpBase {
  Status ValidateDelimiter(const string& delimiter) { return Status::OK(); }
  Status Split(const string& input, const string& delimiter, vector<string>* parts) {
    return str_util::Split(...);
  }  
};

class StringSplitUTF8Op : public StringSplitOp {
  Status ValidateDelimiter(const string& delimiter) { return str_util::ValidUTF8(delimiter); }
  Status Split(...) { return str_util::SplitUTF8(...); }
};
```

That way it's easy to keep a bulk of the logic in sync?

Reply:
Thanks @asimshankar. The class has been split as suggested.